packages/core/src/SessionEventEmitter.ts

```diff
@@ -0,0 +1,256 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+import { EventEmitter } from "events";
+import { EVENTS } from "./constant";
+
+export interface ISessionEventEmitter {
+  events: SessionEventEmitter;
+}
+
+export class SessionEventEmitter extends EventEmitter {
```

Comment:
I'm not convinced this needs to be a class - shouldn't this just be an interface and in the long term the events property is created as `public events: SessionEventEmitter = new EventEmitter()`. Functionally the reflection of properties doesn't seem to do anything.

Reply:
That's an excellent point, I didn't think about verifying whether there was an `EventEmitter` interface along the existing class, and there is. Fixed with e5bed656.